packages/api/lib/writeRecords/write-granules.js

```diff
@@ -462,52 +511,95 @@ const _writeGranuleRecords = async (params) => {
     writeConstraints = true,
   } = params;
   let pgGranule;
+  /**
+   * @type { { status: string, pgGranule: PostgresGranuleRecord } | undefined }
+   */
+  let writePgGranuleResult;
 
   log.info('About to write granule record %j to PostgreSQL', postgresGranuleRecord);
   log.info('About to write granule record %j to DynamoDB', apiGranuleRecord);
 
   try {
     await createRejectableTransaction(knex, async (trx) => {
-      pgGranule = await _writePostgresGranuleViaTransaction({
+      writePgGranuleResult = await _writePostgresGranuleViaTransaction({
         granuleRecord: postgresGranuleRecord,
         executionCumulusId,
         trx,
         granulePgModel,
         writeConstraints,
       });
-
-      // Future: refactor to cover the entire object?
-      // Ensure PG default createdAt value is propagated to DynamoDB/ES
-      // in the case where _writeGranule is called without createdAt set
-      if (!apiGranuleRecord.createdAt) {
-        apiGranuleRecord.createdAt = pgGranule.created_at.getTime();
+      if (writePgGranuleResult.status === 'dropped') {
+        return;
       }
+      pgGranule = writePgGranuleResult.pgGranule;
 
-      await granuleModel.storeGranule(apiGranuleRecord, writeConstraints);
-      await upsertGranule({
-        esClient,
-        updates: apiGranuleRecord,
-        index: process.env.ES_INDEX,
-      }, writeConstraints);
+      if (writeConstraints && isStatusActiveState(pgGranule.status)) {
+        // pgGranule was updated, but with writeConstraints conditions and the granule status is
+        // 'queued' or 'running', so only some values were updated. we need to ensure the correct
+        // values are propagated to Dynamo and ES.
+        // The only values allowed to be updated in the PG granule write under these conditions are
+        // currently status, timestamp, updated_at, and created_at, and the associated execution
+        // as part of the write chain
+        const limitedUpdateApiGranuleRecord = await translatePostgresGranuleToApiGranule({
+          granulePgRecord: pgGranule,
+          knexOrTransaction: trx,
+        });
+
+        await granuleModel.storeGranule(limitedUpdateApiGranuleRecord, writeConstraints);
```

Comment:
I'm concerned this is still here, given https://github.com/nasa/cumulus/pull/2954

We need to look into how we even got back to writing dynamo granules.

Reply:
The above was influenced by the weird diff here - the feature branch at https://github.com/nasa/cumulus/blob/feature/rds-phase-3/packages/api/lib/writeRecords/write-granules.js#L372 doesn't have 

```
await granuleModel.storeGranule(apiGranuleRecord, writeConstraints);
```

We need to update this PR to *not* re-add dynamo writes.

Reply:
This was addressed with the commits following the comment.